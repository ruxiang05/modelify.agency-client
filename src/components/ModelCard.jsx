import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../styles/model-card.scss';
import { SelectModel } from '../contexts/selectModelContext';
import { ReactComponent as UnknownUserIcon } from '../assets/icons/user-circle.svg';

class ModelCard extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { model } = this.props;
    const { email } = model;
    const { selectModel } = this.context;
    selectModel(email);
  }

  render() {
    const { model } = this.props;
    const { name, _id } = model;
    const { selectModel } = this.context;
    const cardContent = (
      <div className="model-card">
        <UnknownUserIcon />
        <p>{name}</p>
      </div>
    );
    return (
      <React.Fragment>
        {selectModel.length ? (
          <button
            className="model-card-action"
            type="submit"
            onClick={this.handleClick}
          >
            {cardContent}
          </button>
        ) : (
          <Link
            to={{
              pathname: `/models/${_id}`,
              state: model,
            }}
          >
            {cardContent}
          </Link>
        )}
      </React.Fragment>
    );
  }
}

ModelCard.contextType = SelectModel;

ModelCard.propTypes = {
  model: PropTypes.object.isRequired,
};
export default ModelCard;
