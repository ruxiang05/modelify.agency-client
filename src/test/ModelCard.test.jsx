import React from 'react';
import { shallow } from 'enzyme';
import PropTypes from 'prop-types';
import ModelCard from '../components/ModelCard';

const contextTypes = {
  selectModel: PropTypes.func,
};

const model = {
  name: 'name',
  _id: '1',
  email: 'email@email.com',
};

const selectModelContext = {
  selectModel: jest.fn(email => email),
};
let wrapper;
beforeAll(() => {
  ModelCard.contextTypes = contextTypes;
});

beforeEach(() => {
  wrapper = shallow(<ModelCard {...model} />, {
    context: selectModelContext,
  });
});

describe('ModelCard', () => {
  it('should render ModelCard', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render link if selectModel is not provided', () => {
    wrapper = shallow(<ModelCard {...model} />, {
      context: { selectModel: () => {} },
    });
    expect(wrapper.find('Link').length).toBe(1);
  });
  it('should select model if clicked', () => {
    const selectModelButton = wrapper.find('button').first();
    selectModelButton.simulate('click');
    expect(selectModelContext.selectModel).toHaveBeenCalled();
  });
});
