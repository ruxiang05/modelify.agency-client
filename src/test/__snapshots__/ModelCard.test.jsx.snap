// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCard should render ModelCard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModelCard
    model={
      Object {
        "_id": "1",
        "email": "email@email.com",
        "name": "name",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <button
        className="model-card-action"
        onClick={[Function]}
        type="submit"
      >
        <div
          className="model-card"
        >
          <ForwardRef />
          <p>
            name
          </p>
        </div>
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="model-card"
        >
          <ForwardRef />
          <p>
            name
          </p>
        </div>,
        "className": "model-card-action",
        "onClick": [Function],
        "type": "submit",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef />,
            <p>
              name
            </p>,
          ],
          "className": "model-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "name",
            },
            "ref": null,
            "rendered": "name",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "button",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <button
          className="model-card-action"
          onClick={[Function]}
          type="submit"
        >
          <div
            className="model-card"
          >
            <ForwardRef />
            <p>
              name
            </p>
          </div>
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="model-card"
          >
            <ForwardRef />
            <p>
              name
            </p>
          </div>,
          "className": "model-card-action",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef />,
              <p>
                name
              </p>,
            ],
            "className": "model-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "name",
              },
              "ref": null,
              "rendered": "name",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "button",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "selectModel": [MockFunction],
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
